vortex-dtype/src/field.rs

```diff
@@ -110,6 +126,62 @@ impl FieldPath {
         self.0 = self.0.iter().skip(1).cloned().collect();
         Some(self)
     }
+
+    /// The dtype, within the given type, to which this field path refers.
+    ///
+    /// Note that a nullable DType may contain a non-nullable DType. This function returns the
+    /// literal nullability of the child.
+    ///
+    /// # Examples
+    ///
+    /// Extract the type of the "b" field from `struct{a: list(struct{b: u32})?}`:
+    ///
+    /// ```
+    /// use std::sync::Arc;
+    ///
+    /// use vortex_dtype::*;
+    /// use vortex_dtype::Nullability::*;
+    ///
+    /// let dtype = DType::Struct(
+    ///     Arc::new(StructFields::from_iter([(
+    ///         "a",
+    ///         DType::List(
+    ///             Arc::new(DType::Struct(
+    ///                 Arc::new(StructFields::from_iter([(
+    ///                     "b",
+    ///                     DType::Primitive(PType::U32, NonNullable),
+    ///                 )])),
+    ///                 NonNullable,
+    ///             )),
+    ///             Nullable,
+    ///         ),
+    ///     )])),
+    ///     NonNullable,
+    /// );
+    ///
+    /// let path = FieldPath::from(vec![Field::from("a"), Field::ElementType, Field::from("b")]);
+    /// let resolved = path.resolve(dtype).unwrap();
+    /// assert_eq!(resolved, DType::Primitive(PType::U32, NonNullable));
+    /// ```
+    pub fn resolve(&self, mut dtype: DType) -> VortexResult<DType> {
+        for field in &self.0 {
+            dtype = match (dtype, field) {
+                (DType::Struct(fields, _), Field::Name(name)) => fields.field(name)?,
+                (DType::List(element_dtype, _), Field::ElementType) => DType::clone(&element_dtype),
+                (other, f) => {
+                    vortex_bail!("FieldPath: invalid index {:?} for DType {:?}", f, other)
+                }
+            }
+        }
+
+        Ok(dtype)
+    }
+
+    /// Does the field referenced by the field path exist in the given dtype?
+    pub fn exists(&self, dtype: DType) -> bool {
```

Comment:
Again, this feels inverted to me

Reply:
Like this is like saying Path::exists(FileSystem) to me

Reply:
I think I lean towards Andrew's conception, so I went with his approach.

Reply:
https://docs.rs/regex/latest/regex/ I think we should follow this

Reply:
so keep as is?